src/globus_cli/commands/timer/_common.py

```diff
@@ -90,3 +107,81 @@ def isoformat_to_local(
     """,
     flags=re.VERBOSE,
 )
+
+
+def parse_timedelta(s: str) -> datetime.timedelta:
+    matches = timedelta_regex.match(s)
+    if not matches:
+        raise ValueError(f"couldn't parse timedelta from string: {s}")
+    groups = {k: int(v) for k, v in matches.groupdict(0).items()}
+    # timedelta accepts kwargs for units up through days, have to convert weeks
+    groups["days"] += groups.pop("weeks", 0) * 7
+    return datetime.timedelta(**groups)
+
+
+def read_csv(
+    file_name: str,
+    fieldnames: List[str],
+    comment_char: str = "#",
+) -> Generator[Dict[str, Union[str, bool]], None, None]:
+    def decomment(f):
+        for row in f:
+            if not row.startswith(comment_char):
+                yield row
+
+    def transform_val(k: str, v: str) -> Union[str, bool]:
+        v = v.strip()
+        if k == "recursive":
+            try:
+                return bool(strtobool(v))
+            except ValueError:
+                # "invalid truth value"
+                click.echo(f"In file {file_name}: couldn't parse {v} as a boolean")
+                sys.exit(1)
+        else:
+            return v
+
+    with open(file_name) as f:
+        reader = DictReader(decomment(f), fieldnames=fieldnames)
+        for row_dict in reader:
+            yield {k: transform_val(k, v) for k, v in row_dict.items()}
+
+
+def require_activated_endpoints(
+    transfer_client: TransferClient,
+    endpoints: List[str],
+    reactivate_if_expires_in: int = 86400,
+):
+    not_activated = []
+    for endpoint in endpoints:
+        try:
+            if not transfer_client.get_endpoint(endpoint).get("activated"):
```

Comment:
This function, and this line in particular, is the main thing I want to revisit in this PR.

There are dedicated APIs and [SDK support](https://github.com/globus/globus-sdk-python/blob/56085ab4aaf6b6d263313799f94e4388f77ed68e/src/globus_sdk/services/transfer/response/activation.py#L7) for handling activation. I'm looking at how [`globus endpoint is-activated`](https://github.com/globus/globus-cli/blob/b16ef4c18fcfbfa1c76136209ba76b3223523480/src/globus_cli/commands/endpoint/is_activated.py#L100) is implemented, but I also think we can get away with the `endpoint_autoactivate` call as our check.

I think we should take the same approach that other CLI commands take:
- attempt autoactivate (with `if_expires_in=...`)
- if `AutoActivationFailed` is returned, treat that as the endpoint not being activated (for most other commands, that's [an error](https://github.com/globus/globus-cli/blob/b16ef4c18fcfbfa1c76136209ba76b3223523480/src/globus_cli/services/transfer/activation.py#L91))

While we're tinkering with this, other changes worth considering...
- Do we need special handling for the activation call failing? I would tend to say "No" until proven otherwise. The Transfer API should give a good error with an informative message, and the CLI already has top-level handling for these errors.
- Rather than suggesting opening the webapp as the only resolution, what about `globus endpoint activate`? Or perhaps code can be shared with [`globus endpoint activate --web`](https://github.com/globus/globus-cli/blob/b16ef4c18fcfbfa1c76136209ba76b3223523480/src/globus_cli/commands/endpoint/activate.py#L280)?